src/main/java/org/broadinstitute/hellbender/utils/bwa/Alignment.java

```diff
@@ -0,0 +1,154 @@
+    private final int suboptimalScore; // for bwa-specific XS tag
+    private final Cigar cigar;     // cigar for alignment (empty if unmapped)
+    private final String mdTag;    // the MD tag
+    private final String xaTag;    // the XA tag
+    private final int mateRefId;   // mate's refId (-1 if unpaired or if mate unmapped)
+    private final int mateRefStart;// mate's reference start (-1 if unpaired or if mate unmapped)
+    private final int templateLen; // inferred template length (0 if unpaired, mate unmapped, or on different ref contigs)
+
+    public Alignment( final int samFlag, final int refId, final int refStart, final int refEnd,
+                      final int seqStart, final int seqEnd, final int mapQual,
+                      final int nMismatches, final int alignerScore, final int suboptimalScore,
+                      final Cigar cigar, final String mdTag, final String xaTag,
+                      final int mateRefId, final int mateRefStart, final int templateLen ) {
+        this.samFlag = samFlag;
+        this.refId = refId;
+        this.refStart = refStart;
+        this.refEnd = refEnd;
+        this.seqStart = seqStart;
+        this.seqEnd = seqEnd;
+        this.mapQual = mapQual;
+        this.nMismatches = nMismatches;
+        this.alignerScore = alignerScore;
+        this.suboptimalScore = suboptimalScore;
+        this.cigar = cigar;
+        this.mdTag = mdTag;
+        this.xaTag = xaTag;
+        this.mateRefId = mateRefId;
+        this.mateRefStart = mateRefStart;
+        this.templateLen = templateLen;
+    }
+
+    public int getSamFlag() { return samFlag; }
+    public int getRefId() { return refId; }
+    public int getRefStart() { return refStart; }
+    public int getRefEnd() { return refEnd; }
+    public int getSeqStart() { return seqStart; }
+    public int getSeqEnd() { return seqEnd; }
+    public int getMapQual() { return mapQual; }
+    public int getNMismatches() { return nMismatches; }
+    public int getAlignerScore() { return alignerScore; }
+    public int getSuboptimalScore() { return suboptimalScore; }
+    public Cigar getCigar() { return cigar; }
+    public String getMDTag() { return mdTag; }
+    public int getMateRefId() { return mateRefId; }
+    public int getMateRefStart() { return mateRefStart; }
+    public int getTemplateLen() { return templateLen; }
+
+    /**
+     * Transforms an unaligned GATKRead into an aligned one, as specified by this Alignment.
+     * This method will not work reliably on a previously aligned read:  stale tag information about the previous
+     * alignment might get left behind.
+     *
+     * Also, BWA does this odd thing, supplying AS and XS tags for unaligned reads.
+     * To produce SAM records that are byte-identical to the ones created by the bwa mem command line,
+     * call this method with justLikeBWA=true.
+     */
+    public GATKRead apply( final GATKRead originalRead, final List<String> refNames, final SAMFileHeader header,
```

Comment:
another name change suggestion: the argument name `justLikeBWA` isn't quite communicating what it might do IMO, so what about changing it to `generateDefaultASandXS` or something similar?